drivers/wifi/nrf700x/CMakeLists.txt

```diff
@@ -51,25 +49,35 @@ zephyr_library_sources(
 	osal/bus_if/bus/qspi/src/qspi.c
 	osal/fw_if/umac_if/src/cmd.c
 	osal/fw_if/umac_if/src/event.c
-	osal/fw_if/umac_if/src/fmac_ap.c
-	osal/fw_if/umac_if/src/fmac_peer.c
-	osal/fw_if/umac_if/src/fmac_util.c
-	osal/fw_if/umac_if/src/fmac_vif.c
 	osal/fw_if/umac_if/src/fmac_api.c
-	osal/fw_if/umac_if/src/rx.c
-	osal/fw_if/umac_if/src/tx.c
 	zephyr/src/shim.c
 	zephyr/src/zephyr_work.c
 	zephyr/src/timer.c
 	zephyr/src/util.c
-	zephyr/src/zephyr_disp_scan.c
 	zephyr/src/zephyr_fmac_main.c
-	zephyr/src/zephyr_net_if.c
-	zephyr/src/zephyr_wpa_supp_if.c
 	zephyr/src/qspi/src/device.c
 	zephyr/src/qspi/src/rpu_hw_if.c
 )
 
+
+zephyr_library_sources_ifdef(CONFIG_NETWORKING
+	zephyr/src/zephyr_net_if.c
+)
+
+zephyr_library_sources_ifdef(CONFIG_WPA_SUPP
+	zephyr/src/zephyr_wpa_supp_if.c
+)
+
+zephyr_library_sources_ifndef(CONFIG_NRF700X_RADIO_TEST
+	osal/fw_if/umac_if/src/rx.c
+	osal/fw_if/umac_if/src/tx.c
```

Comment:
If we have CONFIG_NETWORKING enabled but CONFIG_NRF700X_RADIO_TEST disabled, then build will fail, as zephyr_net_if.c depends on rx.c and tx.c, so, might need to tweak these defines.

Reply:
Presently differentiating only based on CONFIG_NRF700X_RADIO_TEST enabled or not (i.e. took out the conditional include based on CONFIG_NETWORKING).